File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.js
/**
 * @param {number[]} students
 * @param {number[]} sandwiches
 * @return {number}
 */
var countStudents = function (students, sandwiches) {
  const n = students.length;
  const counter = [0, 0];
  let i = 0;

  students.forEach((s) => counter[s]++);

  while (i < n && counter[sandwiches[i]] > 0) counter[sandwiches[i++]]--;

  return n - i;
};

File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        n, i = len(students), 0
        count = [0] * 2
        
        for s in students:
            count[s] += 1
        
        while i < n and count[sandwiches[i]]:
            count[sandwiches[i]] -= 1
            i += 1
        return n - i
        